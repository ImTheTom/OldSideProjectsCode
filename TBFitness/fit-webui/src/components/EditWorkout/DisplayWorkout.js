import React, { Component } from "react";
import { connect } from "react-redux";
import DisplayLift from "./DisplayLift";

class DisplayWorkout extends Component {
	render() {
		let liftData;
		if (this.props.workout.lifts !== undefined) {
			liftData = this.props.workout.lifts.map((lift, index) => (
				<DisplayLift lifts={lift} liftIndex={lift.id} workoutIndex={this.props.workoutIndex} type={this.props.type} key={index}/>
			));
		}
		return (
			<div>
				<div className = "workout-title">{this.props.workout.name}</div>
				{liftData}
			</div>
		);
	}
}


export default connect(undefined, undefined)(DisplayWorkout);