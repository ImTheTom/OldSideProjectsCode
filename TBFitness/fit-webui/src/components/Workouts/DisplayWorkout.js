import React, { Component } from "react";
import { Link } from "react-router-dom";

class DisplayWorkout extends Component {
	render() {
		const url = "/workout/"+this.props.workout.id;
		const editUrl = "/workout/"+this.props.workout.id+"/edit";
		return (
			<div className="link-box">
				<Link className="workout-name"to={url}>{this.props.workout.name}</Link>
				<Link className="workout-edit"to={editUrl}>edit</Link>
			</div>
		);
	}
}

export default DisplayWorkout;