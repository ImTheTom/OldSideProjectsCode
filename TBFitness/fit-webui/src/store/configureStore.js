import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import workout from "../reducers/workout";
import lifts from "../reducers/lifts";
import editWorkout from "../reducers/editWorkout";

const store = createStore(combineReducers({ workout: workout, lifts: lifts, editWorkout: editWorkout }), applyMiddleware(thunk));

export default store;

