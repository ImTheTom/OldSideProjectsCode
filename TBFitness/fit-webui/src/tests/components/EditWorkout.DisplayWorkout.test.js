import DisplayWorkout from "../../components/EditWorkout/DisplayWorkout";
import React from "react";
import { shallow } from "enzyme";
import WorkoutBase from "../fixtures/editWorkoutBase";
import store from "../../store/configureStore";
import { Provider } from "react-redux";

test("should render diplay workout component correctly", () => {
	const wrapper = shallow(
		<Provider store={store}>
			<DisplayWorkout workout={WorkoutBase.current} workoutIndex = {WorkoutBase.current.id} type="Remove" />
		</Provider>
	);
	expect(wrapper).toMatchSnapshot();
});