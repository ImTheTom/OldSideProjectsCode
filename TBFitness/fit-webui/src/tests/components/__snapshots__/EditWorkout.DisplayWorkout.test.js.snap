// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render diplay workout component correctly 1`] = `
<Connect(DisplayWorkout)
  type="Remove"
  workout={
    Object {
      "id": 1,
      "lifts": Array [
        Object {
          "id": 1,
          "name": "BenchPress",
          "reps": 5,
          "sets": 5,
          "user_id": 1,
          "weight": 20,
        },
        Object {
          "id": 2,
          "name": "Overhead Press",
          "reps": 10,
          "sets": 3,
          "user_id": 1,
          "weight": 10,
        },
      ],
      "name": "push day",
      "user_id": 1,
    }
  }
  workoutIndex={1}
/>
`;
